spring-ws-core/src/main/java/org/springframework/ws/transport/http/HttpComponents5MessageSender.java

```diff
@@ -178,7 +182,10 @@ public void setMaxConnectionsPerHost(Map<String, String> maxConnectionsPerHost)
 
 	@Override
 	public void afterPropertiesSet() throws Exception {
-		this.httpClient = clientFactory.getObject();
+
+		if (null != this.clientFactory) {
+			this.httpClient = clientFactory.getObject();
```

Comment:
I'd prefer sticking with the `if (this.clientFactory != null)` paradigm, which you can see in the polishing commit on the main branch.

Reply:
ok